dali/pipeline/data/tensor_vector.h

```diff
@@ -123,368 +92,110 @@ class TensorVector {
   }
 
   auto size() const noexcept {
-    return tensors_.size();
+    return curr_tensors_size_;
   }
 
   size_t ntensor() const noexcept {
-    return tensors_.size();
+    return curr_tensors_size_;
   }
 
-  size_t nbytes() const noexcept {
-    if (state_ == State::contiguous) {
-      return tl_->nbytes();
-    }
-    // else
-    size_t total_nbytes = 0;
-    for (const auto &t : tensors_) {
-      total_nbytes += t->nbytes();
-    }
-    return total_nbytes;
-  }
+  size_t nbytes() const noexcept;
+
+  size_t capacity() const noexcept;
+
+  TensorListShape<> shape() const;
 
-  size_t capacity() const noexcept {
-    if (state_ == State::contiguous) {
-      return tl_->capacity();
-    }
-    // else
-    size_t total_capacity = 0;
-    for (const auto &t : tensors_) {
-      total_capacity += t->capacity();
-    }
-    return total_capacity;
+  TensorShape<> tensor_shape(int idx) const {
+    assert(state_ == State::contiguous && curr_tensors_size_ == views_count_ ||
+           state_ == State::noncontiguous);
```

Comment:
This check isn't really necessary in this function, is it?

Reply:
As you noticed in a comment below, this only makes sense to use if `UpdateViews` was correctly invoked. I wanted to add a debug check here, but I guess this one is incorrect... You have some other idea?